File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/history/id/UnknownRelatedObjectId.js
define([ "stjs", "util/Exceptions", "as/dto/common/id/IObjectId" ], function(stjs, exceptions, IObjectId) {
	var UnknownRelatedObjectId = function(relatedObjectId, relationType) {
		this.relatedObjectId = relatedObjectId;
		this.relationType = relationType;
	};
	stjs.extend(UnknownRelatedObjectId, null, [ IObjectId ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.history.id.UnknownRelatedObjectId';
		constructor.serialVersionUID = 1;
		prototype.relatedObjectId = null;
		prototype.relatedType = null;
		prototype.getRelatedObjectId = function() {
			return this.relatedObjectId;
		};
        prototype.getRelationType = function() {
            return this.relationType;
        };
	}, {});
	return UnknownRelatedObjectId;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/history/id/UnknownRelatedObjectId.java
/*
 * Copyright 2012 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.history.id;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.IObjectId;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonObject("as.dto.history.id.UnknownRelatedObjectId")
public class UnknownRelatedObjectId implements IObjectId
{

    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String relatedObjectId;

    @JsonProperty
    private String relationType;

    public UnknownRelatedObjectId(String relatedObjectId, String relationType)
    {
        this.relatedObjectId = relatedObjectId;
        this.relationType = relationType;
    }

    public String getRelatedObjectId()
    {
        return relatedObjectId;
    }

    public String getRelationType()
    {
        return relationType;
    }

    //
    // JSON-RPC
    //

    protected UnknownRelatedObjectId()
    {
    }

}
